packages/preferences/src/browser/workspace-preference-provider.ts

```diff
@@ -26,9 +26,9 @@ export class WorkspacePreferenceProvider extends AbstractResourcePreferenceProvi
     protected readonly workspaceService: WorkspaceService;
 
     async getUri(): Promise<URI | undefined> {
-        const root = await this.workspaceService.root;
-        if (root) {
-            const rootUri = new URI(root.uri);
+        const workspaceFolder = (await this.workspaceService.roots)[0];
```

Comment:
Should we put a TODO comment to all the locations where we just fetch the first root, so we don't miss them?

Reply:
no worries :) I will review all references of `WorkspaceService` before creating my next PR